packages/crypto/src/crypto.ts

```diff
@@ -0,0 +1,498 @@
+  isCompressed: boolean = true
+): Uint8Array => {
+  return p256.getPublicKey(privateKey, isCompressed);
+};
+/**
+ * HPKE Decrypt Function
+ * Decrypts data using Hybrid Public Key Encryption (HPKE) approach.
+ *
+ * @param {HPKEDecyptParams} params - The decryption parameters including ciphertext, encapsulated key, and receiver private key.
+ * @returns {Promise<Uint8Array>} - The decrypted data.
+ */
+export const hpkeDecrypt = async ({
+  ciphertextBuf,
+  encappedKeyBuf,
+  receiverPriv,
+}: HPKEDecyptParams): Promise<Uint8Array> => {
+  try {
+    let ikm: Uint8Array;
+    let info: Uint8Array;
+    const receiverPubBuf = getPublicKey(base64urlDecode(receiverPriv), false);
+    const aad = additionalAssociatedData(encappedKeyBuf, receiverPubBuf);
+    const kemContext = getKemContext(
+      encappedKeyBuf,
+      uint8arrayToHexString(receiverPubBuf)
+    );
+
+    // Step 1: Generate Shared Secret
+    const dh = await deriveDh(encappedKeyBuf, receiverPriv);
+    ikm = buildLabeledIkm(LABEL_EAE_PRK, dh, SUITE_ID_1);
+    info = buildLabeledInfo(LABEL_SHARED_SECRET, kemContext, SUITE_ID_1, 32);
+    const sharedSecret = await extractAndExpand(
+      new Uint8Array([]),
+      ikm,
+      info,
+      32
+    );
+
+    // Step 2: Get AES Key
+    ikm = buildLabeledIkm(LABEL_SECRET, new Uint8Array([]), SUITE_ID_2);
+    info = AES_KEY_INFO;
+    const key = await extractAndExpand(sharedSecret, ikm, info, 32);
+
+    // Step 3: Get IV
+    info = IV_INFO;
+    const iv = await extractAndExpand(sharedSecret, ikm, info, 12);
+
+    // Step 4: Decrypt
+    const decryptedData = await aesGcmDecrypt(ciphertextBuf, key, iv, aad);
+    return decryptedData;
+  } catch (error) {
+    console.error("Decryption Error:", error);
+    throw error;
+  }
+};
+
+/**
+ * Convert a string to a base64url-encoded string.
+ *
+ * @param {string} input - The input string to encode.
+ * @returns {string} - The base64url-encoded string.
+ */
+export const stringToBase64urlString = (input: string): string => {
+  const base64String = btoa(input);
+  return base64StringToBase64UrlEncodedString(base64String);
+};
+
+/**
+ * Convert a base64 string to a base64url-encoded string.
+ *
+ * @param {string} input - The base64 string to convert.
+ * @returns {string} - The base64url-encoded string.
+ */
+export const base64StringToBase64UrlEncodedString = (input: string): string => {
+  return input.replace(/\+/g, "-").replace(/\//g, "_").replace(/=/g, "");
+};
+
+/**
+ * Decrypt an encrypted credential bundle.
+ *
+ * @param {string} credentialBundle - The encrypted credential bundle.
+ * @param {string} embeddedKey - The private key for decryption.
+ * @returns {Promise<Uint8Array | null>} - The decrypted data or null if decryption fails.
+ */
+export const decryptBundle = async (
+  credentialBundle: string,
+  embeddedKey: string
+): Promise<Uint8Array | null> => {
+  try {
+    let bundleBytes: Uint8Array;
+
+    // Decode using either Base58Check or Base64URL
```

Comment:
I don't think this is needed, you can drop the base64url branch. All bundles are base58check encoded these days!